default_test.go

```diff
@@ -121,28 +163,46 @@ func TestFormatHostname(t *testing.T) {
 		"xn-80ak6aa92e.co",
 		"xn-80ak6aa92e.com",
 		"xn--ls8h.la",
+		"x.",       // valid trailing dot
+		"foo.bar.", // valid trailing dot
+		// extended symbol alphabet
 		"☁→❄→☃→☀→☺→☂→☹→✝.ws",
+		"💩.tv",
 		"www.example.onion",
 		"www.example.ôlà",
 		"ôlà.ôlà",
 		"ôlà.ôlà.ôlà",
-		"ex$ample",
 		"localhost",
```

Comment:
Are you safe against localhost:81 ?

I'm unsure if the code is able to deal with port provided in the URL field you are parsing.

Please note, it might be out of the scope of IsHostname might only receive the host part

 But I prefer to raise the point.

Reply:
Adding it to the tests. Yes it fails. This just validates a host (or domain), not a complete URL.